test/utils.py

```diff
@@ -79,3 +80,83 @@ def wait_file(dir_name, file_pattern, exclude_list, timeout_sec=1):
+
+    Performs this steps for it:
+    1) Copy the instance config files to the run pytest tmp directory.
+       This cfg file should be in /test/integration/foo-module/test_file/instance_cfg_file_name.
+       But you can specify different path by instance_cfg_file_dir arg.
+
+    2) Also copy to pytest tmpdir the lua module utils.lua with the auxiliary functions.
+       This functions is required for using with your instance.
+       As a result, you can use require('utils') inside your instance config file.
+
+    3) Run tarantool via subprocess.Popen with given cfg file.
+       Gets bound port from tmpdir.
+       Init subprocess object and instance's port as attributes.
+
+    NOTE: Demand require('utils').bind_free_port(arg[0]) inside your instance cfg file.
+
+    Args:
+        instance_cfg_file_name (str): file name of your test instance cfg.
+        instance_cfg_file_dir (str): path to dir that contains instance_cfg_file_name.
+        tmpdir (str): expected result of fixture get_tmpdir from conftest.
+
+    Attributes:
+        popen_obj (Popen[bytes]): subprocess.Popen object with tarantool test instance.
+        port (str): port of tarantool test instance.
+
+    Methods:
+        start():
+            Start tarantool test instance and init self.port attribute.
+        stop():
+            Stop tarantool test instance by popen.kill call.
+    """
+
+    def __init__(self, instance_cfg_file_name, instance_cfg_file_dir, tmpdir):
+        # Copy the instance config files to the run pytest tmpdir directory.
+        shutil.copy(instance_cfg_file_dir + "/" + instance_cfg_file_name, tmpdir)
+        # Copy the lua module with the auxiliary functions required by the instance config file.
+        shutil.copy(instance_cfg_file_dir + "/../../../" + "utils.lua", tmpdir)
+
+        self._tmpdir = tmpdir
+        self._instance_cfg_file_name = instance_cfg_file_name
+
+    def start(self):
+        """Start tarantool test instance and init self.port attribute.
+
+        Raises:
+            ValueError:
+                If could not find a file with an instance bound port during 3 seconds deadline.
+                You may have forgotten to use require('utils').bind_free_port(arg[0])
+                inside your cfg instance file.
+        """
+        popen_obj = subprocess.Popen(["tarantool", self._instance_cfg_file_name], cwd=self._tmpdir)
+
+        # Waiting 3 seconds for instance configure itself and dump bound port to temp file.
+        deadline = time.time() + 3
+        while time.time() < deadline:
+            if os.path.exists(str(self._tmpdir) + '/' + self._instance_cfg_file_name + '.port'):
+                break
+            time.sleep(0.1)
+        else:
+            raise ValueError('Could not find a file with an instance bound port')
+
+        # Read bound port of test instance from file in temp pytest directory.
+        file_with_port_path = str(self._tmpdir) + '/' + self._instance_cfg_file_name + '.port'
```

Comment:
Nit: `str(self._tmpdir) + '/' + self._instance_cfg_file_name + '.port'` is written twice. I would store it in a variable.

Reply:
Done